packages/itertools/iter.pony

```diff
@@ -544,6 +564,35 @@ class Iter[A] is Iterator[A]
     ```
     `0 4 1 5 2 3`
     """
+    Iter[A!](
+      object is Iterator[A!]
+        var _use_original: Bool = true
+
+        fun ref has_next(): Bool =>
+          _iter.has_next() or other.has_next()
+
+        fun ref next(): A! ? =>
+          // Oscillate between iterators
+          if _use_original then
+            _use_original = _use_original.op_not()
```

Comment:
The `not` keyword is a unary operator that calls `.op_not()` under the hood. We should probably prefer using the keyword here over using the method directly.
```suggestion
            _use_original = not _use_original
```

Reply:
I started with using `not` but ran into a compilation error which `op_not()` did not have -- no other changes were made. Let me make this change and see if/what error I get.

Reply:
It must have been another error that I read too quickly because I just made the change and no issue.